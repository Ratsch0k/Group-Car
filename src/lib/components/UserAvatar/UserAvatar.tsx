import React from 'react';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import {makeStyles} from '@material-ui/styles';
import {Avatar, useMediaQuery, useTheme} from '@material-ui/core';

/**
 * Props for the UserAvatar.
 */
interface AvatarProps {
  /**
   * Id of the user.
   */
  userId?: number;
  /**
   * Size of the icon.
   */
  size?: 'small' | 'medium' | 'large' | number;
}

/**
 * Dimension of avatar in header bar.
 */
interface UserAvatarDim {
  [index: string]: number;
  small: number;
  medium: number;
  large: number;
}
const avatarDims: UserAvatarDim = {
  small: 30,
  medium: 40,
  large: 50,
};

/**
 * UserAvatar component.
 * If the user is logged in it will display the avatar
 * if the user, if not an icon will be displayed.
 * @param props Props
 */
export const UserAvatar: React.FC<AvatarProps> = (props: AvatarProps) => {
  const {userId, size} = props;
  const theme = useTheme();
  const smallerXs = useMediaQuery(theme.breakpoints.down('sm'));

  // Depending on size in props change width and height
  let dim;
  if (size !== undefined) {
    if (typeof size === 'number') {
      dim = size;
    } else {
      dim = avatarDims[size];
    }
  } else {
    dim = avatarDims.medium;
  }

  const useStyle = makeStyles({
    avatar: {
      width: dim,
      height: dim,
      margin: 'auto',
    },
    icon: {
      fontSize: dim,
      margin: 'auto',
    },
  });
  const classes = useStyle();

  if (userId) {
    return (
      <Avatar
        className={classes.avatar}
        src={`/api/user/${userId}/profile-pic`}
        alt={`user-${props.userId}`}
      />
    );
  } else {
    return (
      <AccountCircleIcon
        className={classes.icon}
        fontSize={smallerXs ? 'small' : 'large'}
      />
    );
  }
};

export default UserAvatar;
