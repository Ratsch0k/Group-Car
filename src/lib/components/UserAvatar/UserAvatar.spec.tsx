import React from 'react';
import {render} from '@testing-library/react';
import UserAvatar from './UserAvatar';
import '@testing-library/jest-dom/extend-expect';

it('renders without userId', () => {
  const {container} = render(<UserAvatar />);
  expect(container).toMatchSnapshot();
});

it('renders with given userId', () => {
  const userId = 12;
  const {container} = render(<UserAvatar userId={userId} />);

  expect(container).toMatchSnapshot();
  expect(container.getElementsByTagName('img')[0])
      .toHaveAttribute('src', `/api/user/${userId}/profile-pic`);
});
