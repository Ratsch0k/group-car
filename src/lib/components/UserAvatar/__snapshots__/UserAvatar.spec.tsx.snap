// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with given userId 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular makeStyles-avatar-3"
  >
    <img
      alt="user-12"
      class="MuiAvatar-img"
      src="/api/user/12/profile-pic"
    />
  </div>
</div>
`;

exports[`renders without userId 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root makeStyles-icon-2 MuiSvgIcon-fontSizeLarge"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
    />
  </svg>
</div>
`;
